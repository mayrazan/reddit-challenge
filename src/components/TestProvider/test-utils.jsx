import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import theme from '../../theme';
import GlobalStyle from '../../theme/GlobalStyle';
import { lightColors } from '../../theme/colors';

const AllTheProviders = ({ children, ...props }) => (
  <ThemeProvider theme={{ ...theme, ...lightColors }} {...props}>
    <GlobalStyle />
    {children}
  </ThemeProvider>
);
AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
