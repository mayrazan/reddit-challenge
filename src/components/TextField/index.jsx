import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const InputWrapper = styled.div`
  width: 100%;
  align-self: center;
  ${({ theme }) => theme.breakpoints.md} {
    width: 50%;
  }
`;

const Input = styled.input`
  width: 100%;
  border: 1px solid ${({ theme }) => theme.background.border};
  padding: 12px 16px;
  outline: 0;
  border-radius: ${({ theme }) => theme.borderRadius};
  box-shadow: 0 10px 15px 0 rgba(111, 30, 81, 0.05);
`;

export default function TextField({
  placeholder,
  name,
  onChange,
  value,
  ...props
}) {
  return (
    <InputWrapper>
      <Input
        placeholder={placeholder}
        name={name}
        onChange={onChange}
        value={value}
        type="search"
        {...props}
      />
    </InputWrapper>
  );
}

TextField.propTypes = {
  placeholder: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};
