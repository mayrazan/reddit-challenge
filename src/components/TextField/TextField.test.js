import React from 'react';
import user from '@testing-library/user-event';
import { render, screen } from '../TestProvider/test-utils';

import TextField from './index';

describe('<TextField />', () => {
  test('renders component', () => {
    render(
      <TextField
        placeholder="buscar posts"
        value="react"
        onChange={() => {}}
        name="busca"
        type="search"
      />,
    );
    const textField = screen.getByPlaceholderText(/buscar posts/i);
    expect(textField).toMatchSnapshot();
  });

  describe('when', () => {
    describe('user is typing', () => {
      test('the value must be updated', () => {
        const onChangeMock = jest.fn();
        render(
          <TextField
            placeholder="Buscar posts"
            value=""
            onChange={onChangeMock}
            name="busca"
            isTouched
          />,
        );

        const inputValue = screen.getByPlaceholderText(/buscar posts/i);
        user.type(inputValue, 'react');
        expect(onChangeMock).toHaveBeenCalledTimes(5);
      });
    });
  });
});
