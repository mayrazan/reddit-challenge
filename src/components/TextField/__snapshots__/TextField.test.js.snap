// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> renders component 1`] = `
.c0 {
  width: 100%;
  border: 1px solid #4C5667;
  padding: 12px 16px;
  outline: 0;
  border-radius: 8px;
  box-shadow: 0 10px 15px 0 rgba(111,30,81,0.05);
}

@media screen and (min-width:768px) {

}

<input
  class="c0"
  name="busca"
  placeholder="buscar posts"
  type="search"
  value="react"
/>
`;
